### PR TITLE
Add PHP 8.3 support

### PR DESCRIPTION
**Pull request by realFlowControl**
### Description

This PR will add PHP 8.3 support to the tracer and profiler and also adds PHP 8.3 to our CI and build process.
Please review #2282 before and merge into this branch

PROF-8311

### Readiness checklist
- [ ] (only for Members) Changelog has been added to the release document.
- [x] Tests added for this feature/bug.

### Reviewer checklist
- [x] Appropriate labels assigned.
- [x] Milestone is set.
- [ ] Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

File: dockerfiles/ci/centos/7/php-8.3/configure.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
+if [ -z "${PHP_SRC_DIR}" ]; then
+    echo "Please set PHP_SRC_DIR"
+    exit 1
+fi
+
+# needed for CentOS7 vs PHP 8.3, otherwise you get a:
+# ld: dynamic STT_GNU_IFUNC symbol `mb_wchar_to_utf16le' with pointer equality in `ext/mbstring/libmbfl/filters/mbfilter_utf16.o' can not be used when making an executable; recompile with -fPIE and relink with -pie
+# see also https://github.com/php/php-src/issues/11603
+export LDFLAGS=-pie
+export CFLAGS=-fPIE
```
Comment (realFlowControl) [inline dockerfiles/ci/centos/7/php-8.3/configure.sh:10]: @bwoebi can you help me with this? I am in the impression, that this should ne be necessary, but it is for CentOS 7 on `x86_64` for PHP 8.3. I do not need those flags for `arm64` or PHP 8.2 or for PHP 8.3 on Ubuntu / `x86_64`.
Can you help me understand whats going on?

Comment (realFlowControl) [reply dockerfiles/ci/centos/7/php-8.3/configure.sh:10]: Thanks @bwoebi for the quick discussion, seems like this is related to CentOS 7 working with an outdated gcc, so we'll leave this there and guard it to only be set on `x86_64`.

**Comment by realFlowControl** [comment]
Rebased this on the latest `master` after merging #2208 and rebuild the image for PHP 8.3 with Rust 1.71, LLVM 16 and updated PHP 8.3 to beta 2

**Comment by realFlowControl** [comment]
Looks like there is PHP 8.3 support for Xdebug, I'll update this PR to include Xdebug
https://phpc.social/@derickr/111018444824055510

**Review by morrisonlevi — APPROVED** [review]
On my side, it looks good.

**Comment by bwoebi** [comment]
Shall we temporarily disable the php_installer tests for 8.3 until it's released?

**Comment by realFlowControl** [comment]
> Shall we temporarily disable the php_installer tests for 8.3 until it's released?

Sounds like a good idea, I'll push the changes later today

**Review by bwoebi — APPROVED** [review]
Time for the merging - thanks for all the work, Florian!

**Comment by morrisonlevi** [comment]
This failure looks interesting: https://app.circleci.com/pipelines/github/DataDog/dd-trace-php/13000/workflows/aad5bebd-6397-4923-be36-593a1fc60955/jobs/2979723. Has it been discussed yet?

**Comment by realFlowControl** [comment]
Should definitely keep an eye on that, interestingly it is succeeding locally with latest `PHP-8.3`

**Pull request merged**